src/codetypes.h

```diff
@@ -55,8 +57,31 @@
 #define HOLEMAPSIZE        192
 // #define NFRINGE            3
 // #define NVAR               6
+#define WALLNODETYPE       0
+#define OUTERNODETYPE      1
 /*==================================================================*/
-/* inline debugging tools                                             */
+/* ADAPTIVE HOLE MAP OCTANT INFO                                    */
+/*==================================================================*/
+//#define USE_ADAPTIVE_HOLEMAP 1 // [0] original hole map, [1] adaptive hole map
+#define INTERSECT_ALG        1 // [0] point-box inclusion only
```

Comment:
In this picture, the point-box algorithm would only tag the cells with the end-point nodes in them, missing the cells with surface intersections (any cell that has a line through it). Here, the middle element and the middle-bottom violate the outer-surface, wall-surface principal. This will cause the hole-map construction to fail at forming a fully-closed hole map.
![Alg-Type](https://user-images.githubusercontent.com/37650539/236029158-3df6ce67-b473-416d-9b70-45fce91742f9.png)